lib/auth/default_current_user_provider.rb

```diff
@@ -336,7 +337,7 @@ def lookup_api_user(api_key_value, request)
       api_username = header_api_key? ? @env[HEADER_API_USERNAME] : request[API_USERNAME]
 
       params = @env['action_dispatch.request.parameters']
-      unless api_key.request_allowed?(request, params)
+      unless api_key.request_allowed?(@env)
```

Comment:
I think we don't need to assign the `params` variable anymore if we pass the env directly.